addons/portal/wizard/portal_wizard.py

```diff
@@ -15,182 +11,43 @@
 # (note that calling '_' has no effect except exporting those strings for translation)
 
 
-def extract_email(email):
-    """ extract the email address from a user-friendly email address """
-    addresses = email_split(email)
-    return addresses[0] if addresses else ''
-
-
-
-class wizard(osv.osv_memory):
+class PortalWizard(models.TransientModel):
     """
         A wizard to manage the creation/removal of portal users.
     """
     _name = 'portal.wizard'
     _description = 'Portal Access Management'
 
-    _columns = {
-        'portal_id': fields.many2one('res.groups', domain=[('is_portal', '=', True)], required=True,
-            string='Portal', help="The portal that users can be added in or removed from."),
-        'user_ids': fields.one2many('portal.wizard.user', 'wizard_id', string='Users'),
-        'welcome_message': fields.text(string='Invitation Message',
-            help="This text is included in the email sent to new users of the portal."),
-    }
-
-    def _default_portal(self, cr, uid, context):
-        portal_ids = self.pool.get('res.groups').search(cr, uid, [('is_portal', '=', True)])
-        return portal_ids and portal_ids[0] or False
+    def _default_portal(self):
+        return self.env['res.groups'].search([('is_portal', '=', True)], limit=1)
 
-    _defaults = {
-        'portal_id': _default_portal,
-    }
+    portal_id = fields.Many2one('res.groups', domain=[('is_portal', '=', True)], required=True, string='Portal', help="The portal that users can be added in or removed from.", default=_default_portal)
+    user_ids = fields.One2many('portal.wizard.user', 'wizard_id', string='Users')
+    welcome_message = fields.Text(string='Invitation Message', help="This text is included in the email sent to new users of the portal.")
 
-    def onchange_portal_id(self, cr, uid, ids, portal_id, context=None):
+    @api.onchange('portal_id')
+    def onchange_portal_id(self):
         # for each partner, determine corresponding portal.wizard.user records
-        res_partner = self.pool.get('res.partner')
-        partner_ids = context and context.get('active_ids') or []
+        ResPartner = self.env['res.partner']
+        partner_ids = self.env.context.get('active_ids')
         contact_ids = set()
         user_changes = []
-        for partner in res_partner.browse(cr, SUPERUSER_ID, partner_ids, context):
+        for partner in ResPartner.sudo().browse(partner_ids):
             for contact in (partner.child_ids or [partner]):
                 # make sure that each contact appears at most once in the list
                 if contact.id not in contact_ids:
                     contact_ids.add(contact.id)
                     in_portal = False
                     if contact.user_ids:
-                        in_portal = portal_id in [g.id for g in contact.user_ids[0].groups_id]
+                        in_portal = self.portal_id in contact.user_ids[0].groups_id.ids
```

Comment:
If you store list of ids into variable then variable name should be with suffix of ```_id```